cpp/include/cudf/lists/detail/gather.cuh

```diff
@@ -288,6 +288,37 @@ std::unique_ptr<column> gather_list_leaf(
   rmm::cuda_stream_view stream,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Segmented gather of the elements within a list element in each row of a list column.
+ *
+ * * @code{.pseudo}
+ * source_column   : [{"a", "b", "c", "d"}, {"1", "2", "3", "4"}, {"x", "y", "z"}]
+ * gather_map_list : [{0, 1, 3, 2}, {1, 3, 2}, {}]
+ *
+ * result          : [{"a", "b", "d", "c"}, {"2", "4", "3"}, {}]
+ * @endcode
+ *
```

Comment:
There should be some words in the documentation about how this only supports lists with depth 1.

Reply:
It supports more depth too. Unit tests cover depth>1.

Reply:
I was referring to the `gather_map`.

Reply:
updated.